### PR TITLE
Updating ServerCommon to 2.94

### PR DESCRIPTION
**Pull request by agr**
Recent update to server common messed up Azure SQL connection factory made it depend on `ICachingSecretInjector`.
Updating DI setup and tests to accommodate.

Also, unrelated change: fix to locale-sensitive test.

Tested the change with Orchestrator.

### CODE REVIEW

File: src/NuGet.Jobs.Common/JsonConfigurationJob.cs [hunk 1]
```diff
@@ -121,14 +121,16 @@ private IConfigurationRoot GetConfigurationRoot(string configurationFilename, ou
...
-        private IServiceProvider GetServiceProvider(IConfigurationRoot configurationRoot, ISecretInjector secretInjector, ISecretReader secretReader)
+        private IServiceProvider GetServiceProvider(IConfigurationRoot configurationRoot, ICachingSecretInjector secretInjector, ICachingSecretReader secretReader)
         {
             // Configure as much as possible with Microsoft.Extensions.DependencyInjection.
             var services = new ServiceCollection();
 
             if (!_validateOnly)
             {
+                services.AddSingleton<ISecretInjector>(secretInjector);
                 services.AddSingleton(secretInjector);
```
Comment (shishirx34) [inline src/NuGet.Jobs.Common/JsonConfigurationJob.cs:132]: Any particular reason to register both `ISecretReader` and `ICachingSecretReader` with DI? Or just for not breaking any instances where `ISecretReader` is used in jobs?

**Review by shishirx34 — APPROVED** [review]

**Comment by joelverhagen** [comment]
Does this affect the secret reloading behavior of Status Aggregator? Might be worth deploying this to DEV and verifying this is still true (or verifying some other way that you know):
https://github.com/NuGet/Engineering/issues/4125#issuecomment-994294479

File: src/NuGet.Jobs.Common/JsonConfigurationJob.cs (hunk 1 above)
Comment (agr) [reply src/NuGet.Jobs.Common/JsonConfigurationJob.cs:132]: `ISecretReader` interface wasn't changed. `ICachingSecretReader` provides some additional capabilities and is needed to create SQL connections. Keeping two registrations will enable new scenarios and not break old ones.

**Comment by agr** [comment]
@joelverhagen deployed, see no change in behavior: secrets are reloaded as before according to logs.

**Pull request merged**